includes/admin/home/help/class-sensei-home-help-provider.php

```diff
@@ -14,25 +14,74 @@ class Sensei_Home_Help_Provider {
 	/**
 	 * Return a list of categories which each contain multiple help items.
 	 *
-	 * @return Sensei_Home_Help_Category[]
+	 * @return array[]
 	 */
 	public function get(): array {
```

Comment:
WDYT about, instead of having a `get()` method, have a `jsonSerialize` method and then implement [`JsonSerializable`](https://www.php.net/manual/en/class.jsonserializable.php)?

That way, technically we don't even need to call `->get()` everywhere, [as PHP would do that for us recursively](https://www.php.net/manual/en/jsonserializable.jsonserialize.php#121377)...

The same thing also applies to all the other classes that implement `get()` and return a JSON serializable array.

Reply:
Hey, that's an interesting idea but generally speaking I prefer to make things more explicit so it's easier to read/follow for us in the future (or other developers that might come).

What I mean is that I see easier to understand seeing a `*Provider->get()` than seeing an actual `*Provider` getting returned and having to know about the `JsonSerializable` feature. But thanks for raising it, because I will let it run in background the possibility of `JsonSeriablizable` for the future.